### DISCUSSION TITLE
getActivePinia was called with no active Pinia. Did you forget to install pinia?

### DISCUSSION THREAD

**Discussion by mJeromeDiaz** [quasarframework/quasar#13947] [General - Components / Directives / etc] [answered]
Hi, 
i search and try a lot since 2 days, but all my try return the same error : getActivePinia was called with no active Pinia. Did you forget to install pinia?

I use quasar 3.3.3 and pinia:  2.0.16, vuex is removed.

Have some idea ?

My boot files (correctly added in quasar.conf.js)
```js
// src/boot/pinia.js
import { boot } from "quasar/wrappers";
import { createPinia } from "pinia";

const pinia = createPinia();

export default boot(async ({ app }) => {
  app.use(pinia);
});
```   
My store
```js
//src/stores/auth.js
import { defineStore } from 'pinia';
import axios from 'axios'

export const authStore = defineStore('authStorage', {
  state: () => {
    return {
      user: {}
    };
  },
  actions: {
    async logUser(){
     console.info('pinia was here')
      await axios({
        method: 'POST',
        url: 'auth/login',
        data: {"username":"","password":""},
        headers: { Accept: "application/json", "Content-Type": "application/json" }
        
      })
      .then(response => {
        console.log(response.data)
      })
    }
  },

});
```   

and my component

```js
import { authStore } from '../../stores/auth.js';
const store_auth= authStore()

export default defineComponent({
  mounted(){
    store_auth.logUser()
  }
});
```

**Comment by TobyMosque** [comment, marked as answer]
just create a new project with pinia and copy the `./stores` dir from the brand new project to the old one.
```
cp -R path/to/new/project/stores path/to/old/project/stores/
```
optional, keep your vuex stores at `./store` and write/port your pinia stores at `./stores`
so, remove your `boot/pinia`

regarding the options API, i would recommend to u, use Pinia with the setup (Composition API), but if u prefer stick with the Options API, u would map your stores.

```js
import { authStore } from 'stores/auth.js';
import { mapStores } from 'pinia';

export default defineComponent({
  mounted () {
    console.log(this.authStorageStore) // store name "authStorage" + the "Store" suffix
    this.authStorageStore.logUser()
  },
  computed: {
    ...mapStores(authStore)
  }
});
```

**Reply by TobyMosque** [reply]
a personal recommendation, u would read about async/await and promise chains, since both does the same, probably u would stick only to one. https://javascript.info/async

promise chain
```js
logUser(){
  console.info('pinia was here')
  return axios({
    method: 'POST',
    url: 'auth/login',
    data: {"username":"","password":""},
    headers: { Accept: "application/json", "Content-Type": "application/json" }
  })
  .then(response => {
    console.log(response.data)
  })
  .catch(err => {
    throw err;
  })
}
```

async/await
```js
async logUser(){
  try {
    const { data } = await axios({
      method: 'POST',
      url: 'auth/login',
      data: {"username":"","password":""},
      headers: { Accept: "application/json", "Content-Type": "application/json" }
    })
    console.log(data)
  } catch (err) {
    throw err;
  }
}
```

**Reply by mJeromeDiaz** [reply]
thank your for your help.

I create a new quasar project, install pinia and remove vuex.
I copy pasted the stores folder into the new project and changed src/pages/index.vue with your code.

i still got the same error.

thank for your recommendation, i'll go to read this :)

**Reply by TobyMosque** [reply]
do u can share u project? or at a project where that error occurs?

**Reply by mJeromeDiaz** [reply]
Sorry, I left from work, but I just recreated a project from home.
i do my homework... 

i have the same error.

like you want, i just copy past stores folder and your code in src/pages/index.vue, i dont create a boot file
https://github.com/mJeromeDiaz/quasar_pinia

**Reply by mJeromeDiaz** [reply]
I modified my work, and added a starter file, because I didn't understand where in your code your pinia creation instance.
And it works well 👍

Thank you, really thank you.
I leave the repository for noobies like me

**Reply by TobyMosque** [reply]
**stores/index.js**
```js
import { store } from 'quasar/wrappers';
import { createPinia } from 'pinia';

export default store((/* { ssrContext } */) => {
  const pinia = createPinia();
  return pinia;
});

```
